### DISCUSSION TITLE
[MVUX] Add ability to execute command on the UI thread

### DISCUSSION THREAD

**Discussion by dr1rrb** [unoplatform/uno.extensions#2417] [Ideas]
By default MVUX will defer all its work on background threads (subscribing to feed, enumerating them and commands), but this can be confusing for users that are not familiar with such concerns.

While MVUX is an opinionated framework and I'm fine that by default everything is executed on background thread, I'm wondering if we should consider to add the ability to request to run some pieces on the UI thread, noticeably commands.

A solution could be to simply OPT-IN using the `CommandAttribute` :

```csharp
[Command(RunOnUiThread: true)]
public void SimpleNavigateToSecondPage()
{
   Frame.Navigate(typeof(MySecondPage));
}
```

**Comment by francoistanguay** [comment]
No. Presentation shouldnt care about UI. Let's leave this responsibility to services. Or use an IDispatcher to always try to schedule on UI thread.
